nssp_demo/postprocess_scoring.R

```diff
@@ -0,0 +1,412 @@
+script_packages <- c(
+  "dplyr",
+  "scoringutils",
+  "lubridate",
+  "ggplot2",
+  "argparser"
+)
+
+## load in packages without messages
+purrr::walk(script_packages, \(pkg) {
+  suppressPackageStartupMessages(
+    library(pkg, character.only = TRUE)
+  )
+})
+
+
+epiweek_to_date <- function(epiweek, epiyear) {
+  # Create date for January 1st of the epiyear
+  jan1 <- as.Date(paste0(epiyear, "-01-01"))
+  # Calculate days to add (epiweeks start on Sunday)
+  days_to_add <- (epiweek - 1) * 7
+  # Add days and adjust to previous Sunday
+  date <- jan1 + days_to_add
+  date <- date - lubridate::wday(date, week_start = 7)
+  return(date)
+}
```

Comment:
I think this is off by one, but it's not 100% clear what the function is supposed to do

``` r
library(tidyverse)
epiweek_to_date <- function(epiweek, epiyear) {
  # Create date for January 1st of the epiyear
  jan1 <- as.Date(paste0(epiyear, "-01-01"))
  # Calculate days to add (epiweeks start on Sunday)
  days_to_add <- (epiweek - 1) * 7
  # Add days and adjust to previous Sunday
  date <- jan1 + days_to_add
  date <- date - lubridate::wday(date, week_start = 7)
  return(date)
}

tibble(date = seq(ymd("2024-06-01"), ymd("2024-06-30"), by =1)) %>% 
  mutate(weekday = wday(date, label = T)) %>% 
  mutate(start_of_epiweek = map_vec(date, ~epiweek_to_date(epiweek(.x), epiyear(.x))))
#> # A tibble: 30 × 3
#>    date       weekday start_of_epiweek
#>    <date>     <ord>   <date>          
#>  1 2024-06-01 Sat     2024-05-25      
#>  2 2024-06-02 Sun     2024-06-01      
#>  3 2024-06-03 Mon     2024-06-01      
#>  4 2024-06-04 Tue     2024-06-01      
#>  5 2024-06-05 Wed     2024-06-01      
#>  6 2024-06-06 Thu     2024-06-01      
#>  7 2024-06-07 Fri     2024-06-01      
#>  8 2024-06-08 Sat     2024-06-01      
#>  9 2024-06-09 Sun     2024-06-08      
#> 10 2024-06-10 Mon     2024-06-08      
#> # ℹ 20 more rows
```

<sup>Created on 2024-11-04 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

Reply:
We should replace it with this https://github.com/cdcent/cfa-forecasttools/blob/main/R/epiweek_to_date.R

Reply:
Done in  52bb233